### ISSUE TITLE
Networking samples do not work on MacOS

### ISSUE THREAD

**Issue by FunMiles** [andreasbuhr/cppcoro#64]
It seems that all the `socket_send_*` and `socket_recv_*` are only defined for Windows.
Did I miss something?

**Comment by andreasbuhr** [comment]
Yes, unfortunately there is only a Windows implementation for networking and file-IO.

**Comment by FunMiles** [comment]
I am short of time to enter into development to implement this. If I end up with more free time, I may try to tackle it for UNIX type sockets (Linux/MacOS) and follow up.

**Comment by andreasbuhr** [comment]
Actually, there is s pull request pending implementing this, see https://github.com/andreasbuhr/cppcoro/pull/63. I did not find the time to review it yet because I moved to a new city and changed to a new role recently.
If you could test this pull request and write a review that would be greatly appreciated.

Be aware that the CI might be broken currently. There were some changes in GitHub actions and this project is not adapted to these changes yet.
